src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -77,92 +78,102 @@ public static IEnumerable<VisualStudioInstance> GetInstances(string workingDirec
+        private static extern int hostfxr_resolve_sdk2(
+            string exe_dir,
+            string working_dir,
+            hostfxr_resolve_sdk2_flags_t flags,
+            hostfxr_resolve_sdk2_result_fn result);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_get_available_sdks(string exe_dir, hostfxr_get_available_sdks_result_fn result);
+
         private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
         {
-            const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
-            
-            Process process;
-            var lines = new List<string>();
-            try
+            string dotnetPath = File.Exists("/usr/share/dotnet/dotnet") ? "/usr/share/dotnet" :
+            File.Exists($"/home/{Environment.GetEnvironmentVariable("USER")}/share/dotnet/dotnet") ? $"/home/{Environment.GetEnvironmentVariable("USER")}/share/dotnet" :
+            File.Exists("dotnet") ? "." :
+            !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DOTNET_HOME")) ? Environment.GetEnvironmentVariable("DOTNET_HOME") :
+            null;
+
+            if (dotnetPath == null)
             {
-                process = new Process()
-                { 
-                    StartInfo = new ProcessStartInfo("dotnet", "--info")
+                // Windows
+                foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(';'))
+                {
+                    if (File.Exists(Path.Combine(dir, "dotnet.exe")))
                     {
-                        WorkingDirectory = workingDirectory,
-                        CreateNoWindow = true,
-                        UseShellExecute = false,
-                        RedirectStandardOutput = true
+                        dotnetPath = dir;
                     }
-                };
-
-                // Ensure that we set the DOTNET_CLI_UI_LANGUAGE environment variable to "en-US" before
-                // running 'dotnet --info'. Otherwise, we may get localized results.
-                process.StartInfo.EnvironmentVariables[DOTNET_CLI_UI_LANGUAGE] = "en-US";
+                }
 
-                process.OutputDataReceived += (_, e) =>
+                // Unix
+                foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(':'))
                 {
-                    if (!string.IsNullOrWhiteSpace(e.Data))
+                    if (File.Exists(Path.Combine(dir, "dotnet")))
                     {
-                        lines.Add(e.Data);
+                        dotnetPath = dir;
                     }
-                };
-
-                process.Start();
-            }
-            catch
-            {
-                // when error running dotnet command, consider dotnet as not available
-                yield break;
+                }
```

Comment:
I asked. I don't think unifying those makes it better. There are three differences between unix and windows here: Path.PathSeparator, the full executable name, and the delimiter in the PATH variable. Having three places each doing those double checks seems much messier to me, given how little code this actually is. Note that we can't split by both PATH delimiters, as that would break up, for instance, C:\anything